doc/coq.rst

```diff
@@ -0,0 +1,494 @@
+  to depend on a ML plugin,
+
+- Your Coq theory can depend on other theories by specifying them in the
+  ``<coq_theories>`` field. Dune will then pass to Coq the corresponding flags
+  for everything to compile correctly [ ``-Q`` ]. As of today, we only support
+  composition with libraries defined in the same scope (that is to say, under
+  the same ``dune-project`` domain). We will lift this restriction in the
+  future. Note that composition with the Coq's standard library is supported,
+  but in this case the ``Coq`` prefix will be made available in a qualified way,
+  since Coq lang 0.2.
+
+- You can enable the production of Coq's native compiler object files by setting
+  ``<coq_native_mode>`` to ``native``. This will pass ``-native-compiler on`` to
+  Coq and install the corresponding object files under ``.coq-native``, when in
+  ``release`` profile. The regular ``dev`` profile will skip native compilation
+  to make the build faster, available since Coq v0.3.
+
+  Please note: support for native compute is **experimental**, and requires Coq
+  >= 8.12.1; moreover, depending libraries *must* be built with ``(mode
+  native)`` for this to work. Also, Coq must be configured to support native
+  compilation. Note that Dune will explicitly disable output of native
+  compilation objects when ``(mode vo)``, even if the default Coq's configure
+  flag enabled it. This will be improved in the future.
+
+Coq Documenation
+~~~~~~~~~~~~~~~~
+
+Given a :ref:`coq-theory` stanza with ``name A``, dune will produce two
+*directory targets*, ``A.html/`` and ``A.tex/``. HTML or LaTeX documenation for
+a Coq theory may then be built by running ``dune build A.html`` or ``dune build
+A.tex`` respectively (if the dune file for the theory is the current directory).
+
+There are also two aliases that can be called, which will build the HTML or
+LaTeX documentation and these are ``@doc`` and ``@doc-latex`` respectively.
+
+.. _include-subdirs-coq:
+
+Recursive Qualification of Modules
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you add:
+
+.. code:: scheme
+
+    (include_subdirs qualified)
+
+to a ``dune`` file, Dune will consider all the modules in the directory and its
+subdirectories, adding a prefix to the module name in the usual Coq style for
+subdirectories. For example, file ``A/b/C.v`` will be module ``A.b.C``.
+
+Limitations
+~~~~~~~~~~~
+
+- ``.v`` files always depend on the native version of Coq / plugins,
+
+.. _limitation-mlpack:
+
+- A ``foo.mlpack`` file must the present in directories of locally defined
+  plugins for things to work. This is a limitation of ``coqdep``. See the
+  `example plugin`_ or the `this template <https://github.com/ejgallego/coq-plugin-template>`_.
+
+Coq Language Version
+~~~~~~~~~~~~~~~~~~~~
+
+The Coq lang can be modified by adding the following to a dune-project file:
+
+.. code:: scheme
+
+    (using coq 0.3)
+
+The supported Coq language versions (not the version of Coq) are:
```

Comment:
The terminology is confusing.

Reply:
Right. So there is a concept called "dune coq language" which records which dune coq stanzas are allowed in a file. If you have a file which says `(using coq 0.3)` but you are using a version of dune that doesn't support that version of the coq language, then you cannot use the dune file.

Reply:
Yes, we can say "coq build language" or `(lang coq 0.3)` etc...